controller/resource.go

```diff
@@ -37,12 +43,57 @@ func (c *ResourceController) Read(ctx *app.ReadResourceContext) error {
 
 // Register runs the register action.
 func (c *ResourceController) Register(ctx *app.RegisterResourceContext) error {
-	// ResourceController_Register: start_implement
 
-	// Put your logic here
+	// TODO validate the PAT here
 
-	// ResourceController_Register: end_implement
-	return nil
+	var res *resource.Resource
+
+	err := application.Transactional(c.db, func(appl application.Application) error {
+
+		// Lookup or create the resource type
+		resourceType, err := appl.ResourceTypeRepository().LookupOrCreate(ctx, ctx.Payload.Name)
+		if (err != nil) {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		// Lookup the parent resource if one has been specified
+		var parentResource *resource.Resource
+		if (ctx.Payload.ParentResourceID != nil) {
+			parentResource, err = appl.ResourceRepository().Load(ctx, *ctx.Payload.ParentResourceID)
+
+			if (err != nil) {
+				return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
Just return err here. You will wrap it as JSONErrorResponse outside of the transaction (see line 93)